idea-plugin/src/com/reason/ide/annotations/ErrorAnnotator.java

```diff
@@ -1,113 +1,229 @@
-    Collection<OutputInfo> collectedInfo = errorsManager.getInfo(filename);
-    if (LOG.isDebugEnabled()) {
-      int size = collectedInfo.size();
-      LOG.debug("Collected info for file " + filename + ": " + size);
-      if (size == 0) {
-        Pair<Set<String>, Set<String>> keys = errorsManager.getKeys();
-        LOG.debug("  Errors:", keys.first);
-        LOG.debug("  Warnings:", keys.second);
-      }
+  public InitialInfo collectInformation(
+      @NotNull PsiFile psiFile, @NotNull Editor editor, boolean hasErrors) {
+    if (hasErrors) {
+      LOG.error("Annotator was initialized with errors. This isn't supported.");
+      return null;
     }
 
-    for (OutputInfo info : collectedInfo) {
-      LogicalPosition start =
-          new LogicalPosition(
-              info.lineStart < 1 ? 0 : info.lineStart - 1, info.colStart < 1 ? 0 : info.colStart);
-      LogicalPosition end =
-          new LogicalPosition(
-              info.lineEnd < 1 ? 0 : info.lineEnd - 1, info.colEnd < 1 ? 0 : info.colEnd);
-      String message = info.message.replace('\n', ' ').replaceAll("\\s+", " ").trim();
+    Project project = psiFile.getProject();
+    VirtualFile sourceFile = psiFile.getVirtualFile();
 
-      int startOffset = editor.logicalPositionToOffset(start);
-      int endOffset = editor.logicalPositionToOffset(end);
-      if (0 < startOffset && 0 < endOffset && startOffset < endOffset) {
-        if (LOG.isDebugEnabled()) {
-          LOG.debug("annotate " + startOffset + ":" + endOffset + " '" + message + "'");
-        }
-        TextRangeInterval range = new TextRangeInterval(startOffset - 1, endOffset - 1);
-        result.add(new BsbErrorAnnotation(info.isError, message, range, start));
-      } else {
-        if (LOG.isDebugEnabled()) {
-          LOG.debug(
-              "Failed to locate info: "
-                  + start
-                  + "->"
-                  + end
-                  + ", offsets "
-                  + startOffset
-                  + "->"
-                  + endOffset
-                  + ", info "
-                  + info);
-        }
+    // create temporary compilation directory, once
+    if (m_compilationDirectory == null) {
+      m_compilationDirectory = createTempCompilationDirectory(project);
+    }
+
+    Optional<VirtualFile> contentRoot = BsPlatform.findContentRootForFile(project, sourceFile);
+    if (!contentRoot.isPresent()) {
+      LOG.info("Unable to find BuckleScript content root.");
+      return null;
+    }
+
+    VirtualFile bsExecutableDirectory = contentRoot.get().findFileByRelativePath("lib/bs");
+    if (bsExecutableDirectory == null) {
+      LOG.info("Unable to find lib/bs for BuckleScript project.");
+      return null;
+    }
+
+    // Read bsConfig to get the jsx value and ppx
+    // @TODO register a file listener for bsconfig.json
+    VirtualFile bsConfigFile = contentRoot.get().findFileByRelativePath("bsconfig.json");
```

Comment:
you could use com.reason.ide.files.BsConfigJsonFileType#FILENAME

Reply:
ah, couldn't find it because it was `private`